autonav_ws/src/autonav_hardware/src/can_node.py

```diff
@@ -0,0 +1,289 @@
+        linear_pid_statistics_msg.sideways_velocity_setpoint = sideways_velocity_setpoint
+
+        self.linearPIDStatisticsPublisher.publish(linear_pid_statistics_msg)
+
+
+    def publish_angular_pid_statistics(self, msg):
+        angular_velocity, angular_velocity_setpoint = struct.unpack("hh", msg.data)
+        angular_pid_statistics_msg = AngularPIDStatistics()
+        angular_pid_statistics_msg.angular_velocity = angular_velocity
+        angular_pid_statistics_msg.angular_velocity_setpoint = angular_velocity_setpoint
+
+        self.angularPIDStatisticsPublisher.publish(angular_pid_statistics_msg)
+
+
+    def publish_motor_statistics_front_motors(self, msg):
+        left_drive_motor_output, left_steer_motor_output, right_drive_motor_output, right_steer_motor_output= struct.unpack("hh", msg.data)
+        motor_statistics_front_motors = MotorStatisticsFrontMotors()
+        motor_statistics_front_motors.left_drive_motor_output = left_drive_motor_output
+        motor_statistics_front_motors.left_steer_motor_output = left_steer_motor_output
+        motor_statistics_front_motors.right_drive_motor_output = right_drive_motor_output
+        motor_statistics_front_motors.right_steer_motor_output = right_steer_motor_output
+
+        self.motorStatisticsFrontMotorsPublisher.publish(motor_statistics_front_motors)
+
+
+    def publish_motor_statistics_back_motors(self, msg):
+        left_drive_motor_output, left_steer_motor_output, right_drive_motor_output, right_steer_motor_output= struct.unpack("hh", msg.data)
+        motor_statistics_back_motors = MotorStatisticsFrontMotors()
+        motor_statistics_back_motors.left_drive_motor_output = left_drive_motor_output
+        motor_statistics_back_motors.left_steer_motor_output = left_steer_motor_output
+        motor_statistics_back_motors.right_drive_motor_output = right_drive_motor_output
+        motor_statistics_back_motors.right_steer_motor_output = right_steer_motor_output
+
+        self.motorStatisticsFrontMotorsPublisher.publish(motor_statistics_back_motors)
+
+
+    def publish_conbus(self, msg):
+        conbus = Conbus()
+        conbus.id = msg.arbitration_id
+        conbus.data = msg.data
+        self.conbusPublisher.publish(conbus)
+
+
+    # subscriber callbacks
+    def on_safety_lights_received(self, msg:SafetyLights):
+        safety_lights_packet = SafetyLightsPacket()
+        safety_lights_packet.autonomous = msg.autonomous
+        safety_lights_packet.mode = msg.mode
+        safety_lights_packet.red = msg.red
+        safety_lights_packet.green = msg.green
+        safety_lights_packet.blue = msg.blue
+        safety_lights_packet.blink_period = msg.blink_period
+
+        data = bytes(safety_lights_packet)
+        can_msg = can.Message(arbitration_id=arbitration_ids["SafetyLightsCommand"], data=data)
+
+        try:
+            self.can.send(can_msg)
+        except can.CanError:
+            pass
```

Comment:
do we want to be passing these can errors or should the user know about them? how often do they occur? is it like a once every couple seconds because communication protocols be like that or is it a big deal and we should change device state and log an error?